### PR TITLE
Feat: Added parallel limiter to prevent sqlalchemy transaction issues

### PR DESCRIPTION
**Pull request by acasajus**

### CODE REVIEW

File: app/parallel_limiter.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+        return decorated
+
+
+def lock(
+    name: Optional[str] = None,
+    max_secs: int = 5,
+    only_when: Optional[Callable[..., bool]] = None,
+):
+    return __InnerLock(name, max_secs, only_when)
```
Comment (nguyenkims) [inline app/parallel_limiter.py:80]: Nits: I wonder if we can put the implementation directly here without using the class?

File: app/dashboard/views/alias_contact_manager.py [hunk 2]
```diff
@@ -231,6 +231,7 @@ def delete_contact(alias: Alias, contact_id: int):
 
 @dashboard_bp.route("/alias_contact_manager/<int:alias_id>/", methods=["GET", "POST"])
 @login_required
+@parallel_limiter.lock(name="contact_creation")
```
Comment (nguyenkims) [inline app/dashboard/views/alias_contact_manager.py:234]: Do you think is there a way to test this in unit test? We can run a redis container when running test.

File: app/parallel_limiter.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    def acquire_lock(self, lock_name: str, lock_value: str) -> bool:
+        retries = 0
+
+        while not lock_redis.storage.set(
+            lock_name, lock_value, ex=self.max_wait_secs, nx=True
+        ):
+            retries += 1
+            if retries > self.max_wait_secs / self.wait_loop_secs:
+                return False
+            time.sleep(self.wait_loop_secs)
```
[the remark this answers is not in the document]
Comment (acasajus) [reply app/parallel_limiter.py:40]: I'd prefer to leave it generic. Right now it's used in alias and coupon, but we may need to use in routes where there are legitimate use-cases. We could return 429 in line 66 though.

File: app/dashboard/views/custom_domain.py [hunk 4]
```diff
@@ -35,7 +35,7 @@ def custom_domain():
                 return redirect(url_for("dashboard.custom_domain"))
 
             if new_custom_domain_form.validate():
-                new_domain = new_custom_domain_form.domain.data.lower().strip()
+                new_domain = new_custom_domain_form.domain.data.ower().strip()
```
[the remark this answers is not in the document]
Comment (acasajus) [reply app/dashboard/views/custom_domain.py:38]: Good catch. I don't know how that got changed.

File: app/parallel_limiter.py (hunk 1 above)
Comment (acasajus) [reply app/parallel_limiter.py:80]: I didn't manage. We need to keep the state (name, max secs and so on) and with nested functions as soon as they were wrapped the state could not be grabbed implicitly. In any case doing it in a class makes the state easier to follow.

File: app/dashboard/views/alias_contact_manager.py (hunk 2 above)
Comment (acasajus) [reply app/dashboard/views/alias_contact_manager.py:234]: Added a basic unit test.

**Pull request merged**